polyfill/test/duration.mjs

```diff
@@ -1065,6 +1065,147 @@ describe('Duration', () => {
+    const totalD2 = {
+      days: d2Nanoseconds / (24 * 3.6e12),
+      hours: d2Nanoseconds / 3.6e12,
+      minutes: d2Nanoseconds / 6e10,
+      seconds: d2Nanoseconds / 1e9,
+      milliseconds: d2Nanoseconds / 1e6,
+      microseconds: d2Nanoseconds / 1e3,
+      nanoseconds: d2Nanoseconds
+    };
+    it('relativeTo is not required for rounding non-calendar units in durations without calendar units', () => {
+      equal(d2.total({ unit: 'days' }), totalD2.days);
+      equal(d2.total({ unit: 'hours' }), totalD2.hours);
+      equal(d2.total({ unit: 'minutes' }), totalD2.minutes);
+      equal(d2.total({ unit: 'seconds' }), totalD2.seconds);
+      equal(d2.total({ unit: 'milliseconds' }), totalD2.milliseconds);
+      equal(d2.total({ unit: 'microseconds' }), totalD2.microseconds);
+      equal(d2.total({ unit: 'nanoseconds' }), totalD2.nanoseconds);
+    });
+    const endpoint = relativeTo.add(d);
+    const options = (unit) => ({ largestUnit: unit, smallestUnit: unit, roundingMode: 'trunc' });
+    const fullYears = 6;
+    const fullDays = endpoint.since(relativeTo, options('days')).days;
+    const fullMilliseconds = endpoint.since(relativeTo, options('milliseconds')).milliseconds;
+    const partialDayMilliseconds = fullMilliseconds - fullDays * 24 * 3.6e6 + 0.005005;
+    const fractionalDay = partialDayMilliseconds / (24 * 3.6e6);
+    const partialYearDays = fullDays - fullYears * 365 + 2;
+    const fractionalYear = partialYearDays / 365 + fractionalDay / 365;
+    const fractionalMonths = ((endpoint.day - 1) * (24 * 3.6e6) + partialDayMilliseconds) / (31 * 24 * 3.6e6);
+
+    const totalResults = {
+      years: fullYears + fractionalYear,
+      months: 66 + fractionalMonths,
+      weeks: (fullDays + fractionalDay) / 7,
+      days: fullDays + fractionalDay,
+      hours: fullDays * 24 + partialDayMilliseconds / 3.6e6,
+      minutes: fullDays * 24 * 60 + partialDayMilliseconds / 60000,
+      seconds: fullDays * 24 * 60 * 60 + partialDayMilliseconds / 1000,
+      milliseconds: fullMilliseconds + 0.005005,
+      microseconds: fullMilliseconds * 1000 + 5.005,
+      nanoseconds: fullMilliseconds * 1e6 + 5005
+    };
+    for (const [unit, expected] of Object.entries(totalResults)) {
+      it(`total(${unit}) = ${expected}`, () => {
+        equal(d.total({ unit, relativeTo }), expected);
+      });
+    }
+    for (const unit of ['microseconds', 'nanoseconds']) {
+      it(`total(${unit}) may lose precision below ms`, () => {
+        assert(d.total({ unit, relativeTo }).toString().startsWith('174373505005'));
+      });
+    }
+    it('balances differently depending on relativeTo', () => {
+      const fortyDays = Duration.from({ days: 40 });
+      equal(fortyDays.total({ unit: 'months', relativeTo: '2020-01-01' }), 1 + 9 / 29);
+      equal(fortyDays.total({ unit: 'months', relativeTo: '2020-02-01' }), 1 + 9 / 31);
```

Comment:
Yep good catch. BTW, this code never actually ran in my tests (that's why I didn't catch it before) because the previous line ran into the RoundDuration backwards/forwards bug.

Reply:
> I added two assertions to the corresponding test for round() for -40 days relative to 2020-01-01 and 2020-02-01, they should both be -1 - 9/31, might be good to add them here?

Good idea. I broke them out into separate `it()` blocks to clearly differentiate negative tests from positive ones. Along those lines, I'm going to add a few more negative-duration tests too in the next checkin.